kythe/go/platform/kzip/kzip.go

```diff
@@ -173,12 +219,20 @@ func (r *Reader) firstIndex(prefix string) int {
 	return n
 }
 
+func pbName(unitName string) string {
+	return unitName + ".pb"
+}
```

Comment:
See my recent comments in #3633; the more I think about it the less I like mixing different formats in the same container, particularly intermingled and distinguished by suffix.

If we want to extend the format we should attempt to do so in a way that avoids silent surprises (such as old decoders only processing a subset of the compilations) and, likely, take the opportunity to add some facility for extension in the future such as an explicit manifest.

Reply:
I agree with your comments there about not directly intermixing pb and non-pb in the same directory.

I'm going to hold off though because maybe there's a magic optimization to make upstream (e.g. replacement jsonpb library).

Reply:
This wasn't hard to do. Done.